### ISSUE TITLE
pallet-assets MultiCurrencyAdapter support

### ISSUE THREAD

**Issue by GopherJ** [open-web3-stack/open-runtime-module-library#607]
currently orml-xtokens binds too tightly to orml-tokens, orml-currencies. It'll be good to support pallet-assets as well. Maybe we can have another MultiCurrencyAdapter for pallet-assets

**Comment by xlc** [comment]
You should be able to implement your own adapter. Moonbeam team is using xtokens with pallet-assets: https://github.com/PureStake/moonbeam/pull/680

**Comment by GopherJ** [comment]
yes @xlc we did that also, I'm wondering shall this be in `orml-xcm-support` or not.

**Comment by xlc** [comment]
Contributions are welcome. Just that it needs to be feature gated to avoid pulling pallet-assets to projects that not using it.
